build.zig

```diff
@@ -118,3 +122,44 @@ pub fn build(b: *Builder) !void {
     });
     debug_step.dependOn(&debug_cmd.step);
 }
+
+fn countLogTest(b: *Builder, dir: []const u8) u32 {
+    var count: u32 = 0;
+
+    const cwd = fs.cwd();
+
+    var src_dir = std.fs.cwd().openDir(dir, .{ .iterate = true }) catch unreachable;
+    defer src_dir.close();
+
+    var it = src_dir.iterate();
+    while (it.next() catch unreachable) |entry| {
+        if (entry.kind == .Directory) {
+            const dir_path = fs.path.join(b.allocator, &[_][]const u8{ dir, entry.name }) catch unreachable;
+            count += countLogTest(b, dir_path);
+        } else if (entry.kind == .File) {
+            // Open the file
+            var file = src_dir.openFile(entry.name, .{}) catch unreachable;
+            const file_len = file.getEndPos() catch unreachable;
+            // Alloc memory for file
+            var buffer = b.allocator.alloc(u8, file_len) catch unreachable;
+
+            _ = file.readAll(buffer) catch unreachable;
+
+            var index: usize = 0;
+            while (index < file_len) {
+                if (std.mem.indexOf(u8, buffer[index..], "log.logTest")) |i| {
+                    // Found a call
+                    count += 1;
+                    index += i + 1;
+                } else {
+                    // Found all
+                    break;
+                }
+            }
+
+            b.allocator.free(buffer);
+            file.close();
+        }
+    }
+    return count;
+}
```

Comment:
This is an interesting idea but I think it's a bit unnecessary. This will fall down if we have two logTest calls but have them conditionally executed so only one will run. Why not have a list in the python file of expected test log regexes and remove them as you find TESTING log messages that match? Then once the list is empty you can exit the python file knowing everything has passed.

Reply:
But would the list of of regex also have this same problem of if there is a log in a condition

Reply:
Have change to remove cases from a list